packages/trader/src/Stores/Modules/Trading/trade-store.ts

```diff
@@ -931,7 +930,7 @@ export default class TradeStore extends BaseStore {
                             });
 
                             // Clear purchase info on mobile after toast box error disappears (mobile_toast_timeout = 3500)
-                            if (isMobile() && this.root_store.common?.services_error?.type === 'buy') {
+                            if (isNewMobile() && this.root_store.common?.services_error?.type === 'buy') {
```

Comment:
Can we remove `isNewMobile` and pass isMobile flag to `processPurchase` function from the UI component with the help of `useDevice`?